### PR TITLE
Add Share button if the browser supports Web Share.

### LINKED ISSUE #1: (installed) mobile games are lonely
When installed on Android as a WebAPK, the app launches in standalone display mode (meaning that it does not show a URL bar). This makes it quite difficult to share the link to a newly-created game.

Screenshot_20201109-113505

Screenshot_20201109-113520

This should be easy to fix (on modern browsers) using the [Web Share API](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share)

### PR DESCRIPTION
**Pull request by glennhartmann**
See https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share,
https://www.w3.org/TR/web-share/.

Mobile screenshot:
https://user-images.githubusercontent.com/570079/99159195-6f623380-26a8-11eb-83c0-004d3a2f03aa.png

Desktop screenshot (of what it would look like if Web Share was enabled
on desktop):
https://user-images.githubusercontent.com/570079/99159210-84d75d80-26a8-11eb-9bbf-e615f99e54cb.png

Contributes to https://github.com/rknoll/draw/issues/1 and
https://github.com/rknoll/draw/issues/19.

https://github.com/rknoll/draw/issues/1 is basically fixed on mobile,
but the share button is hidden on most desktop browsers, as Web Share
is generally a mobile-geared feature.

### CODE REVIEW

File: client/components/GamePage.js [hunk 1]
```diff
@@ -94,7 +106,10 @@ export default () => {
         <Grid container spacing={3} className={classes.wrapper}>
           <Grid item xs={12} sm={2} className={classes.players}>
             <UserList />
-            <Info />
+            <Grid item xs={12} sm={2} className={classes.bottomButtons}>
```
Comment (rknoll) [inline client/components/GamePage.js]: Do we need a Grid element here or can this be a simple `<div>` instead? I think some of the css rules can also be cleaned up, especially `overflowY` which causes issues with the Info button hover.

**Review by rknoll — COMMENTED** [review]
Thanks! I appreciate the work against #19 but I'm not sure individual png's would scale well in this app. What about we separate the share button from that for now until we approach #19 more globally? We could for example swap out the default font to [xkcd-font](https://github.com/ipython/xkcd-font) (see screenshot below) or [humor-sans](https://github.com/tlrobinson/xkcdplot) as Peter suggested. WDYT?

Instead of the png you could just add a regular `<Button>` with a `startIcon` set to `ShareIcon` for now (see [documentation](https://material-ui.com/components/buttons/)). Oh, and an idea for browsers that don't support WebShare, we could open a dialog and let the user copy the url to the clipboard via the Clipboard API?
Screenshot 2020-11-15 at 12 44 15

File: client/components/GamePage.js (hunk 1 above)
Comment (glennhartmann) [reply client/components/GamePage.js]: Yeah, I didn't notice the info button hover was broken. Should be fixed

desktop: 

mobile:

**Comment by glennhartmann** [comment]
> for browsers that don't support WebShare, we could open a dialog and let the user copy the url to the clipboard via the Clipboard API?

sounds good, but in a different PR

File: client/components/ShareButton.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      fontSize: '1rem',
+    },
+  },
+})));
+
+export default () => {
+  const shareData = {url: window.location.href};
+  const classes = useStyles(navigator.canShare && navigator.canShare(shareData))();
+  // TODO: handle navigator.share() failures.
+  return <div className={classes.buttonContainer}>
```
Comment (rknoll) [inline client/components/ShareButton.js]: I don't think we need the extra wrapper here :-) If we just return a `<Button>` with these styles:
```
  button: {
    marginTop: 'auto',
    marginBottom: 'auto',
    display: canShare ? 'inline-flex' : 'none',
    [theme.breakpoints.down('xs')]: {
      position: 'absolute',
      bottom: 12,
      right: 12,
    },
  },
```

File: client/components/GamePage.js [hunk 3]
```diff
@@ -79,6 +80,12 @@ const useStyles = makeStyles((theme) => ({
       maxHeight: '100%',
     },
   },
+  bottomButtons: {
+    display: 'flex',
+    [theme.breakpoints.up('sm')]: {
+      maxWidth: '100%',
```
Comment (rknoll) [inline client/components/GamePage.js]: Do we need this at all?

File: client/components/ShareButton.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2020 Draw authors.
```
Comment (rknoll) [inline client/components/ShareButton.js]: Let's remove this for now, no other file has a license header.

**Review by rknoll — COMMENTED** [review]
A few more cleanups and this is good to go :-)

File: client/components/GamePage.js (hunk 3 above)
Comment (glennhartmann) [reply client/components/GamePage.js]: I thought so, but I guess not, really.

It does change position on desktop if I remove the "display: 'flex'" though:

display: 'flex': 
Untitled

no display: 'flex': 
Untitled1

File: client/components/ShareButton.js (hunk 2 above)
Comment (glennhartmann) [reply client/components/ShareButton.js]: Done.

File: client/components/ShareButton.js (hunk 4 above)
Comment (glennhartmann) [reply client/components/ShareButton.js]: Done.

File: client/components/GamePage.js (hunk 3 above)
Comment (glennhartmann) [reply client/components/GamePage.js]: CSS, how does it work?

Comment (rknoll) [reply client/components/GamePage.js]: Oh, yeah, I was only referring to the `max-width` and the media query part.. I wish I could select code like in Gerrit :-D
Keeping `display: flex` sgtm, maybe even add `flex-wrap: wrap` so it breaks into two lines on narrow screens :-)

**Review by rknoll — APPROVED** [review]
LGTM % display flex if you want :-)

File: client/components/GamePage.js (hunk 3 above)
Comment (glennhartmann) [reply client/components/GamePage.js]: > I wish I could select code like in Gerrit :-D

yeah, that would be nice. I wouldn't be surprised if there's a Chrome extension or something that enables that.

**Pull request merged**